src/promptflow/tests/executor/unittests/_utils/test_thread_utils.py

```diff
@@ -31,10 +32,10 @@ def test_context_manager(self):
         ):
             time.sleep(10)
         logs = s.getvalue().split("\n")
-        for i in range(1, 10):
-            assert (
-                logs[i - 1]
-                == f"Test has been running for {i} seconds, "
-                + "thread None cannot be found in sys._current_frames, "
-                + "maybe it has been terminated due to unexpected errors."
-            )
+        logs = [log for log in logs if log]
```

Comment:
I might run into similar issue before, `StringIO` misses some content; maybe you can try `capsys` or similar pytest fixture to have a try: <https://docs.pytest.org/en/7.4.x/how-to/capture-stdout-stderr.html>